pkgs/applications/terminal-emulators/mlterm/default.nix

```diff
@@ -1,14 +1,82 @@
-{ stdenv, lib, fetchFromGitHub, pkg-config, autoconf, makeDesktopItem, nixosTests
-, libX11, gdk-pixbuf, cairo, libXft, gtk3, vte
-, harfbuzz #substituting glyphs with opentype fonts
-, fribidi, m17n_lib #bidi and encoding
+{ stdenv
+, lib
+, fetchFromGitHub
+, pkg-config
+, autoconf
+, makeDesktopItem
+, nixosTests
+, vte
+# Open Type layout support, (substituting glyphs with opentype fonts)
+, enableOtl ? true
+, harfbuzz # can be replaced with libotf
+, enableBidi ? true
+, fribidi
+, enableM17n ? !stdenv.isDarwin
+, m17n_lib
+, enableSsh2 ? true
 , libssh2 #build-in ssh
-, fcitx5, fcitx5-gtk, ibus, uim #IME
+, enableFcitx ? !stdenv.isDarwin
+, fcitx5, fcitx5-gtk
+, enableIbus ? !stdenv.isDarwin
+, ibus
+, enableUim ? !stdenv.isDarwin
+, uim #IME
 , wrapGAppsHook #color picker in mlconfig
-, Cocoa #Darwin
+, imagelibChoice ? "gdk-pixbuf" # or mlimgloader depending on your bugs of choice
+, gdk-pixbuf
+, gtk3
+, gtk ? gtk3
+# List of gui libraries to use. According to `./configure --help` ran on
+# release 3.9.3, options are: (xlib|win32|fb|quartz|console|wayland|sdl2|beos)
+, guiChoices ? [
+  "xlib"
+  "fb"
+]
```

Comment:
```suggestion
, withXlib
, withFb
```

Reply:
(Though I don't really see a difference between Xorg/Xlib; these should be one flag.)

Reply:
> (Though I don't really see a difference between Xorg/Xlib; these should be one flag.)

Upstream makes that difference, I wonder too why...